resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -715,15 +702,21 @@ private[spark] class Client(
         .flatten
         .toList
         .sortBy(new File(_).getName)
-      throw new SparkException("Cannot upload files with duplicate names via" +
-        s" ${KUBERNETES_DRIVER_UPLOAD_FILES.key}. The following paths have a duplicated" +
-        s" file name: ${pathsWithDuplicateNamesSorted.mkString(",")}")
+      throw new SparkException("Cannot support files with duplicate names in spark.files because " +
+        "the files are placed into the working directory of the driver and executors.  The " +
+        "following paths have a duplicated file name:" +
+        pathsWithDuplicateNamesSorted.mkString(","))
     }
   }
 
-  private def compressFiles(maybeFilePaths: Option[String]): Option[TarGzippedData] = {
+  private def compressUploadableFiles(maybeFilePaths: Option[String]): Option[TarGzippedData] = {
```

Comment:
I'm not entirely sure if it's worth keeping `TarGzippedData` fields as `Option`s anymore. Sure, we save a few bytes of extra data to upload if there aren't any jars, but it's an extra layer of indirection that has to be semantically understood.